src/UI/Implementation/Component/Modal/Interruptive.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/* Copyright (c) 2016 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Modal;
+
+use ILIAS\UI\Component as Component;
+use ILIAS\UI\Component\Button;
+
+/**
+ * @author Stefan Wanzenried <sw@studer-raimann.ch>
+ */
+class Interruptive extends Modal implements Component\Modal\Interruptive
+{
+
+    use ModalHelper;
+
+    /**
+     * @var Button\Primary
+     */
+    protected $cancel_button;
+
+    /**
+     * @var Button\Standard
+     */
+    protected $action_button;
+
+
+    /**
+     * @inheritdoc
+     */
+    public function __construct($title, Component\Component $content)
+    {
+        parent::__construct($title, $content);
+        // An interruptive modal always has a primary cancel button on the right side
+        $this->cancel_button = $this->getCancelButton('Cancel');
```

Comment:
But I think we should get rid of this, especially passing language keys around. This makes it impossible to construct texts using dynamic information.